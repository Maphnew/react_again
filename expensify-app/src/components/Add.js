import React from 'react';

const AddExpensePage = () => (
    <div>
        This is from my Add ExpensePage
    </div>
);

export default AddExpensePage;