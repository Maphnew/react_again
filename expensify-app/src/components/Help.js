import React from 'react';

const HelpExpensePage = () => (
    <div>
        This is from my Help ExpensePage
    </div>
);

export default HelpExpensePage;