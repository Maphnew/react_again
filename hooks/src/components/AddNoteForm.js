import React, {useContext, useState} from 'react';
import NoteContext from '../context/notes-context'

const AddNoteForm = () => {
    const { dispatch } = useContext(NoteContext)
    const [title, setTitle] = useState('')
    const [body, setBody] = useState('')

    const addNote = (e) => {
        e.preventDefault()
        dispatch({
            type: 'ADD_NOTE',
            title,
            body
        })
        setTitle('')
        setBody('')
    }
    return (
        <div>
            <p>Add note</p>
            <form onSubmit={addNote}>
                <input value={title} onChange={(e) => setTitle(e.target.value)} />
                <textarea value={body} onChange={(e) => setBody(e.target.value)}></textarea>
                <button>add note</button>
            </form>
        </div>
    )
}

export { AddNoteForm as default }