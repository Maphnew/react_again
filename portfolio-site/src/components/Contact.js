import React from 'react';

const Contact = () => (
    <div>
        <h2>Contact</h2>
        This is Contact page.
    </div>
);

export default Contact;