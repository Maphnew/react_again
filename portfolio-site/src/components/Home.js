import React from 'react';

const Home = () => (
    <div>
        <h2>HOME</h2>
        This is Home page.
    </div>
);

export default Home