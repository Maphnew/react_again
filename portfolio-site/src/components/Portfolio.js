import React from 'react';
import { Link } from 'react-router-dom'

const Portfolio = () => (
    <div>
        <h2>My Work</h2>
        This is Portfolio page.
        <Link to="/portfolio/1">Item 01</Link>
        <Link to="/portfolio/2">Item 02</Link>
    </div>
);

export default Portfolio