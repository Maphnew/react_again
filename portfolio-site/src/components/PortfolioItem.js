import React from 'react';

const PortfolioItem = (props) => (
    <div>
        <h2>My Portfolio Item</h2>
        <p>ID: {props.match.params.id}</p>
    </div>
);

export default PortfolioItem