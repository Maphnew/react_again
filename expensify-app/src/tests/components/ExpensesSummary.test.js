import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import expenses from '../fixtures/expenses';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary  expenseCount={1} expensesTotal={235}/>)
    expect(toJSON(wrapper)).toMatchSnapshot();
})

test('should render ExpensesSummary correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary  expenseCount={23} expensesTotal={23512322}/>)
    expect(toJSON(wrapper)).toMatchSnapshot();
})