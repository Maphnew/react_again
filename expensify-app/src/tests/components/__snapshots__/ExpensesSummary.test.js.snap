// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $235.00
  </h1>
</div>
`;

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $23,512,322.00
  </h1>
</div>
`;
